crowbar_framework/lib/crowbar/backup/restore.rb

```diff
@@ -0,0 +1,82 @@
+#
+# Copyright 2015, SUSE LINUX Products GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Backup
+    class Restore
+      attr_accessor :backup, :version
+
+      def initialize(backup)
+        @backup = backup
+        @data = @backup.data
+        @version = @backup.version
+      end
+
+      def restore
+        [:crowbar, :knife, :database].each do |component|
+          ret = send(component)
+          return ret unless ret == true
```

Comment:
This still doesn't work correctly. As all of the called function don't return true on success. I was able to successfully restore a cloud5 backup (manually tweaked because of the HOSTNAME vs hostname issue) with this additional change: https://github.com/SUSE-Cloud/crowbar-core/pull/6